orc/orcarm.c

```diff
@@ -1723,6 +1806,79 @@ orc_arm64_emit_mem (OrcCompiler *p, OrcArm64RegBits bits, OrcArm64Mem opcode,
   orc_arm_emit (p, code);
 }
 
+/** memory access instructions (pair)
+ *
+ *    3                   2                   1
+ *  1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |b 0|1 0 1|0| opt |L|     imm7    |   Rt2   |    Rn   |    Rt   |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ */
+
+#define arm64_code_mem_pair(b,opt,L,imm,Rt2,Rn,Rt) (0x40000000 | ((((b)==64)&0x1)<<31) | \
+    (((opt)&0x7)<<23) | (((L)&0x1)<<22) | (((imm)&0x7f)<<15) | (((Rt2)&0x1f)<<10) | \
+    (((Rn)&0x1f)<<5) | ((Rt)&0x1f))
+
+void
+orc_arm64_emit_mem_pair (OrcCompiler *p, OrcArm64RegBits bits, OrcArm64Mem opcode,
+    int opt, int Rt, int Rt2, int Rn, orc_int32 imm)
+{
+  orc_uint32 code;
+
+  static const char *insn_names[3] = {
+    "stp", "ldp"
+  };
+
+  char opt_rn[ARM64_MAX_OP_LEN];
+
+  if (opcode >= sizeof(insn_names)/sizeof(insn_names[0])) {
```

Comment:
This isn't safe enough. Look at this:
```
mzx@kohaku:/tmp$ gcc test2.c
mzx@kohaku:/tmp$ ./a.out 
24/8
mzx@kohaku:/tmp$ cat test2.c
#include <stdio.h>

int main()
{
  const char *strs[3] = {
    "12345", "abcde" };
  printf("%ld/%ld\n", sizeof(strs), sizeof(strs[0]));
}
mzx@kohaku:/tmp$ 
```

Reply:
If you want to use sizeof(x)/sizeof(x[0]) (looks like equivalent to the macro "ARRAY_SIZE") to count the effective number of elements, you shouldn't specify the length of the pointer-array.

Reply:
Oh. Sorry. it's a mistake. I'll remove `3`